SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -817,6 +875,67 @@ void ASpatialFunctionalTest::OnReplicated_bFinishedTest()
 	}
 }
 
+void ASpatialFunctionalTest::LogAndClearRequireHandler()
+{
+	TArray<FSpatialFunctionalTestRequire> RequiresOrdered = RequireHandler.GetAndClearStepRequires();
+
+	ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+	if (FlowController)
+	{
+		const FString& WorkerName = FlowController->GetDisplayName();
+
+		for (const auto& Require : RequiresOrdered)
+		{
+			FString Msg;
+			if (Require.bPassed)
+			{
+				Msg = FString::Printf(TEXT("%s [Passed] %s : %s"), *WorkerName, *Require.Msg, *Require.StatusMsg);
+			}
+			else
+			{
+				Msg = FString::Printf(TEXT("%s [Failed] %s : %s"), *WorkerName, *Require.Msg, *Require.StatusMsg);
+			}
+
+			FlowController->ServerNotifyLogRequireMessages(*Msg, Require.bPassed);
+		}
+	}
+}
+
+void ASpatialFunctionalTest::LogRequireMessages(const FString& Message, bool bPassed)
+{
+	if (bPassed)
+	{
+		UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, Display, TEXT("%s"), *Message);
+		UE_LOG(LogSpatialGDKFunctionalTests, Display, TEXT("%s"), *Message);
+	}
+	else
+	{
+		UE_VLOG(nullptr, LogSpatialGDKFunctionalTests, Error, TEXT("%s"), *Message);
+		UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("%s"), *Message);
+	}
+}
+
+void ASpatialFunctionalTest::MulticastLogRequireMessages_Implementation(const FString& Message, bool bPassed)
+{
+	LogRequireMessages(Message, bPassed);
+}
+
+void ASpatialFunctionalTest::CrossServerLogRequireMessages_Implementation(const FString& Message, bool bPassed)
+{
+	/*
+	 * When running under one process, simply log the message,
+	 * if using multiple-processes, multicast the message so that the Editor process can correctly report a test pass/failure.
+	*/
+	if (HasAuthority() && GIsEditor)
```

Comment:
Again feel this is kind of week indicator of multiprocess, at the very least I would wrap this into a function.

Reply:
Yes I agree that we are actually relying on some previous guarantee that this function will only be called from the Server and that is why we were using this kind of check. I think we might be able to solve this by simply getting the LevelEditorPlaySettings actually, I will look into it.

Reply:
Resolved by adding the  IsRunningUnderOneProcess function

Reply:
hm, this check doesn't seem to have changed?